crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -2812,9 +2829,10 @@ impl Forks {
                         new.push(fork);
                         continue;
                     }
-                    let not_markers = markers.negate();
+
+                    let not_markers = simplify_python(markers.negate(), python_requirement);
                     let mut new_markers = markers.clone();
-                    new_markers.and(fork.markers.negate());
+                    new_markers.and(simplify_python(fork.markers.negate(), python_requirement));
```

Comment:
Honestly, I can't figure out if the `simplify_python` here is necessary. It's definitely necessary on `not_markers` above... But I'm not sure about here. I don't think any tests change when I remove it.

Reply:
Logically, I think this simplification here is right and even necessary. Because while `fork.markers`, by construction, should already be simplified, negating that here might wind up with a marker that includes (or is completely excluded by) `requires-python`. So re-doing the simplification does indeed seem correct.

I think that the same simplification ought to happen to `markers` too right? Specifically, before passing it to `fork.intersect` below.

Reply:
I thought `markers` was already simplified because we simplify when generating `PubGrubDependency`.

Reply:
I.e., here: https://github.com/astral-sh/uv/pull/6076/files#diff-6be6d80fe4821c47b70a372260f55e73b8da8182b8dcad7525d5cd3eb584532bR1523

Reply:
Talking over DM, the markers here are already simplified.